getting-started/src/main/kotlin/com/google/cloud/kotlin/emojify/EmojifyController.kt

```diff
@@ -0,0 +1,149 @@
+        Emoji.ANGER to stream("emojis/anger.png"),
+        Emoji.SURPRISE to stream("emojis/surprise.png"),
+        Emoji.SORROW to stream("emojis/sorrow.png"),
+        Emoji.NONE to stream("emojis/none.png")
+    )
+
+    fun stream(blobName: String): BufferedImage {
+        val strm: InputStream = Channels.newInputStream(storage.reader(bucketName, blobName))
+        return ImageIO.read(strm)
+    }
+
+    fun errorResponse(status: Int, msg: String): EmojifyResponse {
+        log.severe(msg)
+        return EmojifyResponse(status = status, errorMessage = msg)
+    }
+
+    @GetMapping("/emojify")
+    fun emojify(@RequestParam(value = "objectName") objectName: String): EmojifyResponse {
+
+        if (objectName.contains('/')) return errorResponse(400, "Slashes are intentionally forbidden in objectName.")
+        val bucket = storage.get(bucketName) ?: return errorResponse(500, "storage.bucket.name is missing in application.properties.")
+        val publicUrl: String =
+            "https://storage.googleapis.com/$bucketName/emojified/emojified-$objectName" // api response
+
+        val blob = bucket.get(objectName) ?: return errorResponse(400, "Blob specified doesn't exist in bucket.")
+
+        val imgType = blob?.contentType?.substringAfter('/') ?: return errorResponse(400, "Unable to read source image ContentType from GCS.")
+
+        // Setting up image annotation request
+        val source = ImageSource.newBuilder().setGcsImageUri("gs://$bucketName/$objectName").build()
+        val img = Image.newBuilder().setSource(source).build()
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+            .addFeatures(feat)
+            .setImage(img)
+            .build()
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(listOf(request))
+        if (response.responsesList.size != 1) return errorResponse(500, "Vision API didn't return a single response.")
+        val resp = response.responsesList[0]
+        if (resp.hasError()) return errorResponse(500, resp.error.message)
+
+        // Writing source image to InputStream
+        val imgBuff = stream(objectName)
+        val gfx = imgBuff.createGraphics()
+
+        for (annotation in resp.faceAnnotationsList) {
+            val imgEmoji = emojiBufferedImage[bestEmoji(annotation)]
+            val poly = Polygon()
+            for (vertex in annotation.fdBoundingPoly.verticesList) {
+                poly.addPoint(vertex.x, vertex.y)
+            }
+            val height = poly.ypoints[2] - poly.ypoints[0]
+            val width = poly.xpoints[1] - poly.xpoints[0]
+            // Draws emoji on detected face
+            gfx.drawImage(imgEmoji, poly.xpoints[0], poly.ypoints[1], height, width, null)
+        }
+        // Writing emojified image to OutputStream
+        val baos = ByteArrayOutputStream()
```

Comment:
nit: I think `baos` could use a better name - maybe something a little more descriptive of what it is? (`out`, `output`, `outStream` maybe?)